# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xf86-video-nsc/no-ddc-option.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- xf86-video-nsc-2.8.3/src/amd_lx_driver.c.vanilla	2008-10-21 16:12:40.000000000 +0200
+++ xf86-video-nsc-2.8.3/src/amd_lx_driver.c	2008-10-21 16:23:21.000000000 +0200
@@ -497,7 +497,8 @@
    }
 
    if( (options & PROBE_DETECT) != 0 ) {
-      ConfiguredMonitor = LXProbeDDC(pScrni, pGeode->pEnt->index);
+      if (!xf86ReturnOptValBool(GeodeOptions, OPTION_NO_DDC, FALSE))
+         ConfiguredMonitor = LXProbeDDC(pScrni, pGeode->pEnt->index);
       return TRUE;
    }
 
--- xf86-video-nsc-2.8.3/src/nsc_gx2_driver.c.vanilla	2008-10-21 16:18:30.000000000 +0200
+++ xf86-video-nsc-2.8.3/src/nsc_gx2_driver.c	2008-10-21 16:20:27.000000000 +0200
@@ -490,7 +490,8 @@
    }
 
    if (flags & PROBE_DETECT) {
-      GX2ProbeDDC(pScreenInfo, pGeode->pEnt->index);
+      if (!xf86ReturnOptValBool(GeodeOptions, OPTION_NO_DDC, FALSE))
+         GX2ProbeDDC(pScreenInfo, pGeode->pEnt->index);
       return TRUE;
    }
 #if INT10_SUPPORT
--- xf86-video-nsc-2.8.3/src/nsc_driver.c.vanilla	2008-10-21 16:11:40.000000000 +0200
+++ xf86-video-nsc-2.8.3/src/nsc_driver.c	2008-10-21 16:12:24.000000000 +0200
@@ -324,6 +324,7 @@
    {OPTION_CUSTOM_MODE, "CustomMode", OPTV_BOOLEAN, {0}, FALSE},
    {OPTION_FP_DEST_GEOM, "FPDestGeom", OPTV_ANYSTR, {0}, FALSE},
    {OPTION_FP_ACTIVE_GEOM, "FPActiveGeom", OPTV_ANYSTR, {0}, FALSE},
+   {OPTION_NO_DDC, "NoDDC", OPTV_BOOLEAN, {0}, FALSE},
    {-1, NULL, OPTV_NONE, {0}, FALSE}
 };
 
--- xf86-video-nsc-2.8.3/src/nsc_gx1_driver.c.vanilla	2008-10-21 16:16:10.000000000 +0200
+++ xf86-video-nsc-2.8.3/src/nsc_gx1_driver.c	2008-10-21 16:18:28.000000000 +0200
@@ -485,7 +485,8 @@
    }
 
    if (flags & PROBE_DETECT) {
-      GX1ProbeDDC(pScreenInfo, pGeode->pEnt->index);
+      if (!xf86ReturnOptValBool(GeodeOptions, OPTION_NO_DDC, FALSE))
+        GX1ProbeDDC(pScreenInfo, pGeode->pEnt->index);
       return TRUE;
    }
 
--- xf86-video-nsc-2.8.3/src/nsc.h.vanilla	2008-10-21 16:10:43.000000000 +0200
+++ xf86-video-nsc-2.8.3/src/nsc.h	2008-10-21 16:11:16.000000000 +0200
@@ -540,7 +540,8 @@
    OPTION_CRTENABLE,
    OPTION_CUSTOM_MODE,
    OPTION_FP_DEST_GEOM,
-   OPTION_FP_ACTIVE_GEOM
+   OPTION_FP_ACTIVE_GEOM,
+   OPTION_NO_DDC
 }
 GeodeOpts;
 
